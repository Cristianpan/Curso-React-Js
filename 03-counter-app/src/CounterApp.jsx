import PropTypes from "prop-types";
import { useState } from "react";
export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => {
    setCounter((c) => c + 1);
  };

  const handleSubtrac = () => {
    setCounter((c) => c - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter}</h2>
      <button onClick={handleSubtrac}>-1</button>
      <button aria-label="btn-reset" onClick={handleReset}>Reset</button>
      <button onClick={handleAdd}>+1</button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
