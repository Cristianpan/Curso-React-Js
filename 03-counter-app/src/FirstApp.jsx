import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {    
  return (
    <>
      <h1>{title}</h1>
      <p>{subtitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}
//no usar porque ser√° removido en versiones superiores
//mejor usar los parametros por defecto de js
FirstApp.defaultProps = {
  title: 'No hay titulo', 
  subtitle: 'No hay subtitulo'
}
