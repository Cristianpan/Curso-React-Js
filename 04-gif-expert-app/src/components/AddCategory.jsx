import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onAddCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const handleChange = ({ target }) => {
    const { value } = target;
    setInputValue(value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length <= 1) return;

    onAddCategory({
      id: Date.now(),
      name: inputValue.trim(),
    });
    setInputValue("");
  };

  return (
    <form aria-label="form" onSubmit={handleSubmit}>
      <input
        type="text"
        placeholder="Buscar Gifs"
        value={inputValue}
        onChange={handleChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired,
};
