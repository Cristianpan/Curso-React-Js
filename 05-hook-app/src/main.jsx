import React from "react";
import ReactDOM from "react-dom/client";
import "./index.css";
import { MemoHook } from "./06-memo/MemoHook";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <MemoHook />
  </React.StrictMode>
);
