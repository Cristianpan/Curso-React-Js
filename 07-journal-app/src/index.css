.box-shadow {
    box-shadow: 0px 10px 15px 7px rgba(0,0,0,0.1);
}

.text-animation, .loader-animation {
    animation: faded 2s infinite; 
}

.clip-text {
    display: -webkit-box;
    -webkit-box-orient: vertical;
    overflow: hidden;
}

.clip-text-1 {
    line-clamp: 1;
    -webkit-line-clamp: 1;
}
.clip-text-2 {
    line-clamp: 2;
    -webkit-line-clamp: 2;
}

@keyframes faded {
    0% {
        opacity: 1;
    }

    50% {
        opacity: 0.5;
    }

    100% {
        opacity: 1;
    }
}
