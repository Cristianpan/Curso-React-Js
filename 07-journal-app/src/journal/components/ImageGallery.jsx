import { ImageList, ImageListItem } from "@mui/material";

export const ImageGallery = ({ images }) => {
  return (
    <ImageList sx={{ width: "100%", height: 500 }} cols={4} rowHeight={200}>
      {images.map((img) => (
        <ImageListItem key={img}>
          <img src={img} alt="Imagen de la nota" loading="lazy" />
        </ImageListItem>
      ))}
    </ImageList>
  );
};
