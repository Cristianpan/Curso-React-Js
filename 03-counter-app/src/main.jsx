import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import "./styles.css";
import { FirstApp } from "./FirstApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp title="Cristian" subtitle="Bienvenido a mi pÃ¡gina" /> */}
    <CounterApp
      value={0}
    />
  </React.StrictMode>
);
