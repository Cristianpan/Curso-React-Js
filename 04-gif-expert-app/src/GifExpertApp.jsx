import { useState } from "react";
import { GifGrid, AddCategory } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (newCategory) => {
    if (!categories.some((category) => category.name === newCategory.name)) {
      setCategories([newCategory, ...categories]);
    }
  };

  return (
    <>
      <h1>GifExpertApp</h1>

      <AddCategory onAddCategory={onAddCategory} />
      {!categories.length && <p>¡Agrega una categoría!</p>}

      {categories.map(({ id, name }) => (
        <GifGrid key={id} category={name} />
      ))}
    </>
  );
};
