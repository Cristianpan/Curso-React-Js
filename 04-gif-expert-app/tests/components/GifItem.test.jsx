import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components";

describe("Test to GifItem component", () => {
  const title = "Baki";
  const url =
    "https://media4.giphy.com/media/wOl12QqaDzgUyXsLXj/giphy.gif?cid=ede762f7kisxjc8ocr691xqrwpe63g4fex5hoao7sg5v097x&ep=v1_gifs_search&rid=giphy.gif&ct=g";
  test("Debe de hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("debe de mostrar el titulo en el componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
