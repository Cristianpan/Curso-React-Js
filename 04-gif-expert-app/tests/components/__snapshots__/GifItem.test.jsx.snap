// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test to GifItem component Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Baki"
      src="https://media4.giphy.com/media/wOl12QqaDzgUyXsLXj/giphy.gif?cid=ede762f7kisxjc8ocr691xqrwpe63g4fex5hoao7sg5v097x&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Baki
    </p>
  </div>
</div>
`;